packages/form-control/tests/form-control.test.tsx

```diff
@@ -222,3 +223,95 @@ test("has the correct data attributes", async () => {
   expect(label).toHaveAttribute("data-invalid")
   expect(label).toHaveAttribute("data-readonly")
 })
+
+test("inherits variant styles", async () => {
+  render(
+    <FormControl data-testid="control" id="name" variant="floating-label">
+      <FormLabel data-testid="label">Name</FormLabel>
+      <Input placeholder="Name" />
+    </FormControl>,
+    {
+      wrapper: ({ children }) => (
+        <ChakraProvider
+          theme={extendTheme({
+            components: {
+              Form: {
+                variants: {
+                  "floating-label": {
+                    container: {
+                      bg: "white",
+                      border: "1px",
+                      borderColor: "#ccc",
+                      borderRadius: "base",
+                      h: "3.75rem",
+                      overflow: "hidden",
+                      transition:
+                        "all 500ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
+                      width: "full",
+                    },
+                  },
+                },
+              },
+            },
+          })}
+        >
+          {children}
+        </ChakraProvider>
+      ),
+    },
+  )
+
+  const formControl = screen.getByTestId("control")
+  expect(formControl).toMatchInlineSnapshot(`
```

Comment:
This will fail fairly often because the generated snapshot doesn't match _only_ the variant styles, but also includes styles from `baseStyle` and other pieces of the theming system. 

How do you feel about removing this test completely? We don't test any other components in this way.

Reply:
That is a good point regarding the other parts of the theming system. Yes, I am happy to remove this completely, I was following the advice from the contributing guide, "All commits that fix bugs or add features need a test" but I did also notice that these sort of tests do no exist elsewhere.

Reply:
Sorry about that! That advice is true for many things, but "All" is probably not the right word.